core/src/impl/Kokkos_Half_FloatingPointWrapper.hpp

```diff
@@ -23,6 +23,22 @@
 #include <iosfwd>  // istream & ostream for extraction and insertion ops
 #include <string>
 
+namespace Kokkos::Experimental {
+/// @brief templated struct for determining if half_t is an alias to float.
+/// @tparam T The type to specialize on.
+template <class T>
+struct kokkos_type_is_half_t {
+  enum : bool { value = false };
+};
+
+/// @brief templated struct for determining if bhalf_t is an alias to float.
+/// @tparam T The type to specialize on.
+template <class T>
+struct kokkos_type_is_bhalf_t {
+  enum : bool { value = false };
+};
```

Comment:
Does this need to be user-facing?
```suggestion
/// @brief templated struct for determining if half_t is an alias to float.
/// @tparam T The type to specialize on.
template <class T>
struct kokkos_type_is_half_t : std::false_type {};

/// @brief templated struct for determining if bhalf_t is an alias to float.
/// @tparam T The type to specialize on.
template <class T>
struct kokkos_type_is_bhalf_t : std::false_type {};
```

Reply:
I think it would be helpful to make this user-facing. The current approach of checking the HALF_T_IS_FLOAT macro and then using std::is_same is arduous and messy.